conftest.py

```diff
@@ -3,67 +3,72 @@
 """
 # pylint: disable=redefined-outer-name, unused-argument
 import pytest
-from sqlalchemy.orm import sessionmaker
+from sqlalchemy import create_engine
+from sqlalchemy.engine import Connection, Engine
 from dependency_injector import providers
 from src.app import create_app
 from src.config.container import Container
 from src.config.database import TestingDatabase
 from src.config.db import database_url_connection
 
-database = TestingDatabase()
-engine = database.get_engine()
-Session = sessionmaker()
 container = Container()
 
 
 @pytest.fixture(scope="session")
-def database_instance():
+def engine():
     """
-    Returns database fake instance
+    Creates engine for testing
     """
-    return database
+    return create_engine(url=database_url_connection({"database_name": "oipie_tests"}))
 
 
-@pytest.fixture(scope="module")
-def connection():
+@pytest.fixture(scope="session")
+def connection(engine: Engine):
     """
     Manages database connection
     """
-    connection = database.get_engine().connect()
+    connection = engine.connect()
     yield connection
     connection.close()
 
 
-@pytest.fixture(scope="function")
-def session():
+@pytest.fixture(scope="session")
+def database(connection: Connection):
     """
-    Manages database session and rollsback executed queries
+    Returns database fake instance
     """
-    session = database.session()
-    session.begin_nested()
-    yield session
-    session.rollback()
-    session.close()
+    return TestingDatabase(connection)
 
 
 @pytest.fixture(scope="session", autouse=True)
-def initialise_db():
+def initialise_db(database: TestingDatabase):
     """
     Migrates the database once before running tests
     """
     database.create_database()
+    yield
+    database.drop_database()
```

Comment:
We agreed this is not acceptable.
performance problems and concurrence problems will occur.

Reply:
The scope of the fixture is `session`, so this is only executed before and after *all* test suites

Reply:
that is what we don't want.
we don't want scripts being executed on production.

you could add that script on a different execution than test.
if the test fails is something wrong in your local setup.
I ma not comfortable with dropping all tables when launching tests even when we launch tests on an empty database.